### DISCUSSION TITLE
Need help with visual debugging.

### DISCUSSION THREAD

**Discussion by ghost** [jsvine/pdfplumber#705] [Q&A]
Hello. I am on Windows 10 using VSCode. Python 3.10. I installed MagickWand (and set home path) and GhostScript, but I am not able to get it working. I am receiving this error:
```
. . . \.venv\lib\site-packages\wand\image.py", line 9306, in __init__
    wand = library.NewMagickWand()
OSError: exception: access violation writing 0x0000000000000008
```

Here is the code I am trying to run:
```
import pdfplumber

with pdfplumber.open("416891.pdf") as pdf:
    first_page = pdf.pages[0]
    im = first_page.to_image()
    im.save(f"{first_page.page_number}.png", format="PNG")
```

I'm not necessarily trying to save the image. Really just trying to troubleshoot and look around. Not sure how to actually use MagickWand and GhostScript or if it's supposed to do it automatically.

**Comment by jsvine** [comment]
Hi @donbfry, and thanks for your interest in this library. Based on the helpful error you've shared, the issue seems to stem from [Wand](https://docs.wand-py.org/), the dependency we use in `pdfplumber` to help with visual debugging. The integration is *supposed* to be seamless (requiring no knowledge of Wand), but in this case it seems Wand is having trouble even initializing. 

Looking through Wand's repository, I see a couple of potentially related issues: 

- https://github.com/emcconville/wand/issues/586
- https://github.com/emcconville/wand/issues/499

I wonder: Do either of the solutions suggested there work for you?

**Reply by jsvine** [reply]
It seems that Wand has been updated to address this issue: https://github.com/emcconville/wand/issues/587

If you run `pip install -U wand`, does that resolve the error for you?

**Reply by ghost** [reply]
I apologize for the late reply. I don’t believe this resolves my issue. Although, I am able to save a PNG of the .pdf file now. But I am still unable to get the window to show.

**Reply by ghost** [reply]
I should probably provide some more Information. I am on Windows 11 using VSCode and Python 3.10.6. Wand is at 0.6.10 which should be the latest.

I am trying to run your California WARN report demo. I have tried reinstalling both ImageMagick and Ghostscript. Environment variables are set up correctly as well. I have also tried running the demo code inside and outside of a venv virtual environment. There are no errors but nothing happens.

**Reply by jsvine** [reply]
Hi @donbfry, and thanks for the additional details. Can you explain a bit more about what you mean by "unable to get the window to show"? Do you mean to having the image automatically open on your system (rather than having to navigate to where you saved it and open it from there)? Or perhaps something else?

**Reply by ghost** [reply]
@jsvine I am unsure if I am understanding how to use the visual debugging correctly but I am assuming from the examples that when using things like `im.debug_tablefinder()` that some window would pop up showing the image along with how pdfplumber is interpreting the pdf. Is this correct? Or is that something that is specific to jupyter?

**Reply by jsvine** [reply]
Hi @devwithdon, and you've landed on a not-uncommon confusion that I'm actively trying to clarify. In fact, I've added a bit more documentation and a convenience method in the `develop` branch to address this. (See, e.g., https://github.com/jsvine/pdfplumber/discussions/715.) In the meantime, outside of Jupyter, you can call `im.annotated.show()` (or in your particular example: `im.debug_tablefinder().annotated.show()`) to pop open the image in your local image viewer.
